src/integrationTest/java/com/mongodb/hibernate/EmbeddableIntegrationTests.java

```diff
@@ -0,0 +1,310 @@
+        });
+        loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithFlattenedValues.class, item.id));
+        assertEquals(updatedItem, loadedItem);
+    }
+
+    @Test
+    void nestedValues() {
+        var item = new ItemWithNestedValues();
+        item.id = new StructAggregateEmbeddableValue();
+        item.id.a = 1;
+        item.nested1 = new StructAggregateEmbeddableValue();
+        item.nested1.a = 2;
+        item.nested2 = new StructAggregateEmbeddablePairValue1(3, new StructAggregateEmbeddablePairValue2(4, 5));
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find())
+                .containsExactly(new BsonDocument()
+                        // Hibernate ORM flattens `item.id` despite it being of an aggregate type
+                        .append(ID_FIELD_NAME, new BsonInt32(item.id.a))
+                        .append("nested1", new BsonDocument("a", new BsonInt32(2)))
+                        .append(
+                                "nested2",
+                                new BsonDocument()
+                                        .append("a", new BsonInt32(3))
+                                        .append(
+                                                "nested",
+                                                new BsonDocument()
+                                                        .append("a", new BsonInt32(4))
+                                                        .append("b", new BsonInt32(5)))));
+        var loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithNestedValues.class, item.id));
+        assertEquals(item, loadedItem);
+        var updatedItem = sessionFactoryScope.fromTransaction(session -> {
+            var result = session.find(ItemWithNestedValues.class, item.id);
+            result.nested1.a = -result.nested1.a;
+            return result;
+        });
+        loadedItem = sessionFactoryScope.fromTransaction(session -> session.find(ItemWithNestedValues.class, item.id));
+        assertEquals(updatedItem, loadedItem);
+    }
+
+    @Test
+    void nestedEmptyValue() {
+        var item = new ItemWithNestedEmptyValue();
+        item.id = 1;
+        item.omitted = new StructAggregateEmbeddableEmptyValue();
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertThat(mongoCollection.find()).containsExactly(
+                // Hibernate ORM does not store `item.omitted` despite it being non-`null`
+                new BsonDocument(ID_FIELD_NAME, new BsonInt32(item.id)));
+        var loadedItem =
+                sessionFactoryScope.fromTransaction(session -> session.find(ItemWithNestedEmptyValue.class, item.id));
+        // the entity we stored and the entity we loaded are not equal because Hibernate ORM omits `item.omitted`
+        assertNotEquals(item, loadedItem);
```

Comment:
The behavior mentioned in the comments looks like another Hibernate ORM bug.

Reply:
This seems not a Hibernate ORM bug. It has a property (`false` by default) to configure whether to persist empty embeddable: `hibernate.create_empty_composites.enabled`. 

For instance, when I added the following `hibernate-testing` setting to this class:
```
@ServiceRegistry( settings = @Setting( name = AvailableSettings.CREATE_EMPTY_COMPOSITES_ENABLED, value = "true" ) )
```
you can see all the empty embeddables are working as expected.

Reply:
could we remove those `emptyValue` testing cases for they are working as expected? If we retain them, at least we'd better add more accurate comments not to mislead code readers into thinking that they are Hibernate ORM's bug.

For instance, the following comment:
```
// Hibernate ORM does not store `item.omitted` despite it being non-`null`
```
at least could be changed to
```
// By default Hibernate ORM does not store `item.omitted` despite it being non-`null`
```

Reply:
This configuration property is deprecated, and the feature it controls was removed in Hibernate ORM 7.0:
- https://hibernate.atlassian.net/browse/HHH-11936
- https://lists.jboss.org/archives/list/hibernate-issues@lists.jboss.org/message/6LEDPQL2UI7HPPIQHLQV4UMB2MDGBWEC/

The current behavior is what Hibernate ORM considers the right one, i.e., it's not considered to be a bug.

Reply:
but given currently we are still using v6.6, could we at least mentioned such detail in comment (like since v7, ....) to avoid confusion and be more accurate?

Reply:
Done in fe6190f7bf71108d2b7aeef16b23c7607c4e1c5a. I changed the wording and linked to https://hibernate.atlassian.net/browse/HHH-11936, in case the reader needs more details. I don't want to link to the 6.6 documentation in this case, because if we change the links after upgrading to 7.0, the link won't work. And if we don't change the links, then linking to the documentation of 6.6 from a product that requires 7.0 is going to be weird.